pandas/core/frame.py

```diff
@@ -231,6 +231,21 @@ class DataFrame(NDFrame):
     >>> df3 = DataFrame(np.random.randn(10, 5),
     ...                 columns=['a', 'b', 'c', 'd', 'e'])
 
+    Leaving dtype=None in constructor will infer a wider type than necessary
```

Comment:
What do you mean by an example passing multiple types? Is that possible to pass in multiple types?